src/app/dataset-view/additional-components/flows-component/flows.component.ts

```diff
@@ -0,0 +1,162 @@
+import { ChangeDetectionStrategy, Component, EventEmitter, Input, OnInit, Output } from "@angular/core";
+import {
+    DatasetBasicsFragment,
+    DatasetFlowType,
+    DatasetKind,
+    FlowConnectionDataFragment,
+    FlowDataFragment,
+    FlowStatus,
+    InitiatorFilterInput,
+} from "src/app/api/kamu.graphql.interface";
+import { DatasetFlowsService } from "./services/dataset-flows.service";
+import { Observable, filter, map } from "rxjs";
+import { MaybeNull, MaybeUndefined } from "src/app/common/app.types";
+import { BaseComponent } from "src/app/common/base.component";
+import { NavigationEnd, Router, RouterEvent } from "@angular/router";
+import { requireValue } from "src/app/common/app.helpers";
+import ProjectLinks from "src/app/project-links";
+import { NavigationService } from "src/app/services/navigation.service";
+import { DatasetViewTypeEnum } from "../../dataset-view.interface";
+import { SettingsTabsEnum } from "../dataset-settings-component/dataset-settings.model";
+
+@Component({
+    selector: "app-flows",
+    templateUrl: "./flows.component.html",
+    styleUrls: ["./flows.component.scss"],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class FlowsComponent extends BaseComponent implements OnInit {
+    @Input() public datasetBasics: DatasetBasicsFragment;
+    @Output() onPageChangeEmit = new EventEmitter<number>();
+    public searchFilter = "";
+    public tileWidgetData$: Observable<MaybeUndefined<FlowConnectionDataFragment>>;
+    public flowConnectionData$: Observable<MaybeUndefined<FlowConnectionDataFragment>>;
+    public filterByStatus: MaybeNull<FlowStatus> = null;
+    public filterByInitiator = "All";
```

Comment:
If this is referring to the constant in the table model, than we are in trouble. Looks like we need an enum.